addons/mail/static/tests/owl/hooks/use_store_compare_keys_tests.js

```diff
@@ -0,0 +1,143 @@
+odoo.define('mail.hooks.useStoreCompareKeysTests', function (require) {
+'use strict';
+
+const useStoreCompareKeys = require('mail.hooks.useStoreCompareKeys');
+
+const { Component, QWeb, Store } = owl;
+const { xml } = owl.tags;
+
+const {
+    afterNextRender,
+    nextAnimationFrame,
+} = require('mail.messagingTestUtils');
+
+QUnit.module('mail.messaging', {}, function () {
+QUnit.module('hooks', {}, function () {
+QUnit.module('useStoreCompareKeysTests', {
+    beforeEach() {
+        const qweb = new QWeb();
+        this.env = { qweb };
+        this.start = () => {
+            this.store = new Store({
+                env: this.env,
+                state: {},
+            });
+            this.env.store = this.store;
+        };
+    },
+    afterEach() {
+        this.env = undefined;
+        this.store = undefined;
+    }
+});
+
+
+QUnit.test("compare keys, no depth, primitives", async function (assert) {
+    assert.expect(7);
+    this.start();
+    let count = 0;
+    this.store.state.obj = {
+        subObj1: 'a',
+        subObj2: 'b',
+        use1: true,
+    };
+    class MyComponent extends Component {
+        constructor() {
+            super(...arguments);
+            this.storeProps = useStoreCompareKeys((state, props) => {
+                return {
+                    res: state.obj.use1 ? state.obj.subObj1 : state.obj.subObj2,
+                };
+            }, {
+                onUpdate: () => {
+                    count++;
+                },
+            });
+        }
+    }
+    MyComponent.env = this.env;
+    MyComponent.template = xml`<div t-esc="storeProps.res"/>`;
+
+    const fixture = document.querySelector('#qunit-fixture');
+
+    const myComponent = new MyComponent();
+    await myComponent.mount(fixture);
+    assert.strictEqual(count, 0, 'initial state');
```

Comment:
unclear assertion comment.
suggestion: _"should not detect an update in storeProps initially"_
(this comment is applied on most assertions in this file)

Reply:
ok